modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -932,14 +933,31 @@ private void processValueAttributeOfOperation(SCIMResourceTypeSchema schema, Pat
     }
 
     private List<Map<String, String>> transformMembersAttributeToMap(MultiValuedAttribute multiValuedMembersAttribute)
-            throws CharonException {
+            throws CharonException, BadRequestException {
 
         List<Map<String, String>> memberList = new ArrayList<>();
         List<Attribute> subValuesList = multiValuedMembersAttribute.getAttributeValues();
         for (Attribute subValue : subValuesList) {
             ComplexAttribute complexAttribute = (ComplexAttribute) subValue;
             Map<String, Attribute> subAttributesList = complexAttribute.getSubAttributesList();
 
+            // Check if `value` (member id) attribute is present and not empty.
+            // Check if `value` (member id) attribute is present and not empty.
+            if (!subAttributesList.containsKey(SCIMConstants.CommonSchemaConstants.VALUE)) {
```

Comment:
Let's merge the 1st and 3rd if blocks using an || operator

Reply:
Merged 1st and 3rd blocks in this [commit](https://github.com/wso2/charon/pull/387/commits/4e69ab2724ac31a6e76d13164632d4d896ea115b)